contracts/RenPool.sol

```diff
@@ -70,16 +77,44 @@ contract RenPool {
         uint senderBalance = balances[sender];
 
         require(senderBalance > 0 && senderBalance >= _amount, "Insufficient funds");
-        require(isLocked == false, "Pool is locked");
+        if(!isLocked){
+            totalPooled -= _amount;
+            balances[sender] -= _amount;
+            renToken.transfer(sender, _amount);
+            
+            emit RenWithdrawal(sender, _amount);
+
+        }
+        else{
+            // Pool is locked, withdraw will be put in the queue
+            withdrawRequests.push(WithdrawRequest(msg.sender, _amount));           
+        }
 
-        renToken.transfer(sender, _amount);
-        totalPooled -= _amount;
-        balances[sender] -= _amount;
 
-        emit RenWithdrawal(sender, _amount);
     }
 
     function balanceOf(address _addr) external view returns(uint) {
         return balances[_addr];
     }
+
+    function fullfillWithdrawRequest(uint _withdrawId) external payable{
+        // If pool is locked, look if there is a withdraw queue
+        require(isLocked);
+        require(withdrawRequests.length > 0);
+
+        WithdrawRequest memory withdrawRequest = withdrawRequests[_withdrawId];
+
+        require(renToken.transferFrom(msg.sender, address(this), withdrawRequest.amount));
+
```

Comment:
Interesting, why this is wrapped around require? I've seen this in some other code bases but didn't have the time to search for it

Reply:
I think it is safer to use it this way but I'm really not sure. I've seen it in other codes as well and this is why I've done it like so